OsmAnd/src/net/osmand/aidl/OsmandAidlService.java

```diff
@@ -88,16 +89,31 @@
 
 	private static final String DATA_KEY_RESULT_SET = "resultSet";
 
-	private Map<Long, IOsmAndAidlCallback> callbacks;
+	public static final int KEY_ON_UPDATE = 1;
+	public static final int KEY_ON_NAV_DATA_UPDATE = 2;
+
+	private static Map<Long, AidlCallbackParams> callbacks;
 	private Handler mHandler = null;
 	HandlerThread mHandlerThread = new HandlerThread("OsmAndAidlServiceThread");
 
-	private long updateCallbackId = 0;
+	private static long aidlCallbackId = 0;
 
 	private OsmandApplication getApp() {
 		return (OsmandApplication) getApplication();
 	}
 
+	public static void addAidlCallback(IOsmAndAidlCallback callback, long id, long key) {
+		callbacks.put(id, new AidlCallbackParams(callback, key));
+	}
+
+	public static boolean removeAidlCallback (long id){
```

Comment:
no static

Reply:
removed those methods and add listener. Setting service's listener to aidl api in onCreate